okio/src/jvmMain/kotlin/okio/JvmFileSystemFileHandle.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio
+
+import java.nio.ByteBuffer
+import java.nio.channels.SeekableByteChannel
+
+internal class JvmFileSystemFileHandle(
+  readWrite: Boolean,
+  private val seekableByteChannel: SeekableByteChannel,
+) : FileHandle(readWrite) {
+
+  @Synchronized
+  override fun protectedResize(size: Long) {
+    val currentSize = size()
+    val delta = size - currentSize
+    if (delta > 0) {
+      protectedWrite(currentSize, ByteArray(delta.toInt()), 0, delta.toInt())
+    } else {
+      seekableByteChannel.truncate(size)
+    }
+  }
+
+  @Synchronized
+  override fun protectedSize(): Long {
+    return seekableByteChannel.size()
+  }
+
+  @Synchronized
+  override fun protectedRead(
+    fileOffset: Long,
+    array: ByteArray,
+    arrayOffset: Int,
+    byteCount: Int,
+  ): Int {
+    seekableByteChannel.position(fileOffset)
+    val byteBuffer = ByteBuffer.wrap(array)
```

Comment:
Need to use arrayOffset here. Otherwise you write at a potentially incorrect offset into the array.

Reply:
fixed!